overlord/devicestate/devicemgr.go

```diff
@@ -148,31 +151,82 @@ func (m *DeviceManager) ensureOperational() error {
 		return nil
 	}
 
+	// conditions to trigger device registration
+	//
+	// * have a model assertion with a gadget (core and
+	//   device-like classic) in which case we need also to wait
+	//   for the gadget to have been installed though
+	// * classic with a model assertion with a non-default store specified
```

Comment:
That sounds like a better give a way. Do we still need the former considering this?